### PR TITLE
MT-16-Topic 6 home work

### PR DESCRIPTION
**Pull request by AliakseiDziardziyashenka**
Created new branch for home work ,added 2 new projects

### CODE REVIEW

File: MentoringHomeWork/HW6/HW6.Task1/Program.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿// See https://aka.ms/new-console-template for more information
+
+using HW6.Task1;
+
+var arrayCalculateSum = new ArrayCalculateSum();
+ArrayCalculateSum.ArraySum();
```
Comment (batonkarton) [inline MentoringHomeWork/HW6/HW6.Task1/Program.cs]: You used the static method of the ArrayCalculateSum.ArraySum();
It's not necessary to initialize a new variable and object for this above - you don't use it actually but you used memory.

File: MentoringHomeWork/HW6/HW6.Task1/ArrayCalculateSum.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+namespace HW6.Task1;
+
+public class ArrayCalculateSum
```
Comment (batonkarton) [inline MentoringHomeWork/HW6/HW6.Task1/ArrayCalculateSum.cs]: Naming: ArrayCalculator, ArrayService, ArraySumCalculator

File: MentoringHomeWork/HW6/HW6.Task2/Program.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿// See https://aka.ms/new-console-template for more information
+
+using HW6.Task2;
+
+NewArrayPosition ar = new NewArrayPosition();
+NewArrayPosition.AddNewPosition();
```
Comment (batonkarton) [inline MentoringHomeWork/HW6/HW6.Task2/Program.cs]: The same - you created an object and don't use it - just call the static method.

File: MentoringHomeWork/HW6/HW6.Task2/NewArrayPosition.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
+namespace HW6.Task2;
+
+public class NewArrayPosition
```
Comment (batonkarton) [inline MentoringHomeWork/HW6/HW6.Task2/NewArrayPosition.cs]: Naming - let's try to rename it on your own.

File: MentoringHomeWork/HW6/HW6.Task2/Program.cs (hunk 3 above)
Comment (batonkarton) [inline MentoringHomeWork/HW6/HW6.Task2/Program.cs]: image
I didn't enter the number '0' - could you try to resolve it?

File: MentoringHomeWork/HW6/HW6.Task2/NewArrayPosition.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        }
+
+        numbers[position] = newValue;
+
+        ShowResultArray(numbers);
+    }
+
+    private static void ShowResultArray(int[] numbers)
+    {
+        Console.WriteLine("Yours numbers is : ");
```
Comment (batonkarton) [inline MentoringHomeWork/HW6/HW6.Task2/NewArrayPosition.cs]: Yours numbers ARE: - minor
By the way, great work that you split repeated code in a separate method - well done!

**Review by batonkarton — CHANGES_REQUESTED** [review]
Well done!

Please, fix the review comments.
Let's try to finish 3rd Task too.

File: MentoringHomeWork/HW6/HW6.Task2/NewArrayPosition.cs (hunk 5 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW6/HW6.Task2/NewArrayPosition.cs]: resolved

File: MentoringHomeWork/HW6/HW6.Task2/Program.cs (hunk 3 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW6/HW6.Task2/Program.cs]: resolved

File: MentoringHomeWork/HW6/HW6.Task2/NewArrayPosition.cs (hunk 4 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW6/HW6.Task2/NewArrayPosition.cs]: resolved

File: MentoringHomeWork/HW6/HW6.Task2/Program.cs (hunk 3 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW6/HW6.Task2/Program.cs]: resolved

File: MentoringHomeWork/HW6/HW6.Task1/ArrayCalculateSum.cs (hunk 2 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW6/HW6.Task1/ArrayCalculateSum.cs]: resolved

File: MentoringHomeWork/HW6/HW6.Task1/Program.cs (hunk 1 above)
Comment (AliakseiDziardziyashenka) [reply MentoringHomeWork/HW6/HW6.Task1/Program.cs]: resolved

File: MentoringHomeWork/HW6/HW6.Task2/NewArrayPosition.cs [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
+    public void AddNewPosition()
     {
         int[] numbers = new int[6];
-        
+
         for (int i = 0; i < numbers.Length - 1; i++)
         {
             Console.WriteLine("Please enter the number: ");
             numbers[i] = Convert.ToInt32(Console.ReadLine());
+
```
Comment (batonkarton) [inline MentoringHomeWork/HW6/HW6.Task2/NewArrayPosition.cs]: not necessary new line here

File: MentoringHomeWork/HW6/HW6.Task2/Program.cs [hunk 7]
```diff
@@ -2,5 +2,5 @@
 
 using HW6.Task2;
 
-NewArrayPosition ar = new NewArrayPosition();
-NewArrayPosition.AddNewPosition();
\ No newline at end of file
+ArrayService ar = new ArrayService();
```
Comment (batonkarton) [inline MentoringHomeWork/HW6/HW6.Task2/Program.cs]: bad naming for variable - 'ar' - what is it? let's rename it to make it more reasonable
when you add a name for the variable you should always think about other developers who will read this code after your implementation - is this name informative and understandable for other guys and me after 5 years?

File: MentoringHomeWork/HW6/HW6.Task3/Program.cs [hunk 8]
```diff
@@ -0,0 +1,13 @@
+﻿int[] arrayInt = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 };
+
+for (int i = 0; i < arrayInt.Length / 2; i++)
+{
+    var saveNumber = arrayInt[i];
+    arrayInt[i] = arrayInt[arrayInt.Length - 1 - i];
+    arrayInt[arrayInt.Length - 1 - i] = saveNumber;
+}
+
+foreach (var t in arrayInt)
```
Comment (batonkarton) [inline MentoringHomeWork/HW6/HW6.Task3/Program.cs]: 'var t' - not a good name for this variable - better practice to name it more informative - for example 'number' or 'arrayValue' to make it more reasonable and readable.

**Review by batonkarton — CHANGES_REQUESTED** [review]

File: MentoringHomeWork/HW6/HW6.Task2/Program.cs [hunk 9]
```diff
@@ -2,5 +2,5 @@
 
 using HW6.Task2;
 
-ArrayService ar = new ArrayService();
-ar.AddNewPosition();
\ No newline at end of file
+ArrayService arrayHandler = new ArrayService();
```
Comment (batonkarton) [inline MentoringHomeWork/HW6/HW6.Task2/Program.cs:5]: Not sure about 'handler' but let's leave it as it is.

**Review by batonkarton — APPROVED** [review]

**Pull request merged**